### ISSUE TITLE
Dynamic content interface (Creating custom post types and custom fields in Elementor)

### ISSUE THREAD

**Issue by MichaelDarko** [elementor/elementor#11126]
## Prerequisites

- [x] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [x] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

## Description

I think it's time for Elementor to become a **professional tool for building dynamic websites.**

I know there is some support for ACF and dynamic content, but it's half baked and anytime you try building a dynamic website with multiple post types you have to install many other plugins (Ele Custom Skin etc.) to get it working.

Elementor could eliminate the need for plugins for creating CPT and CF by giving us an awesome interface to create CPT and CF directly in Elementor admin.

## Two strategies

There are two strategies Elementor can choose for complex dynamic websites:

**1. Relying on other plugins (ACF etc.) and supporting their features.**

**2. Becoming a universal tool for creating all kind of websites.**

I think the second strategy would make Elementor much more powerful, future-proof tool and Elementor could build on this functionality for years to come and have everything under one roof.

Building websites with different post types than articles is too complicated for most users. You have to install CPT plugin, ACF plugin, set everything up, then pull everything in Elementor etc.

Most users will simply not bother to go through this and will give up. If this was all possible to do in Elementor, it would change everything. 

## Jet engine

There is one third party plugin that is trying to do all this.

https://crocoblock.com/plugins/jetengine/

The UX is still pretty complicated, but when you learn how to use it, it is extremely powerful. 

To be honest, I'm shocked we're getting such powerful core functionality form third party plugins and Elementor is not doing much on this front. 

I strongly believe that similar interface for creating CPT and CF should be in Elementor Pro.

## Elementor should be WebFlow for WordPress

WebFlow is an amazing tool and creating different "post types" with different fields is beautifuly simple. No need to wory about other plugins, compatibility etc. 

So I think Elementor should basically strive to be WebFlow for Wordpress and let us do all of this inside Elementor.

## Epilogue

You could say having third party solutions is enough. But I don't think so. This is probably the most fundamental functionality outside the basic visual page building. 

We can't have a situation where third party plugins are providing complex core functionality and Elementor is releasing surface level widgets and features. It should be the other way round.

Other page builders and tools like WebFlow are advancing very quickly. If Elementor doesn't decide to become a tool for creating all kinds of websites, it's not a good future-proof strategy.

**Comment by MichaelDarko** [comment]
And to sum up my feature request, Elementor should have an interface to do this:

1. Create new post type
2. Add different meta fields for that post type
3. Design listing widget/archive/single for that post type

This would make Elementor 90% more powerful and future-proof. Relying on other plugins and tools for this is not a good strategy.

**Comment by MichaelDarko** [comment]
I would also like to add a thought about **first principles**.

When you forget about WordPress and think from first principles about what you need to create a website. You will end up with two main things:

**1. Visual page building**
**2. Post types and meta fields**

With these two aspects you can fulfil almost any request from your client. 

As it stands, visual page building is no problem. But anything more complex than that, you have to install many plugins, stich everything together and hope for compatibility.

If Elementor wants to **solve page building**, you have to have an integrated interface for creating post types and meta fields.

**Comment by Matherone** [comment]
Yes, please. Offering assisted creation of **custom post types** as a **native Elementor feature** could create a new major source of **usability value** for Elementor. It could be start of something new. Even going stand-alone.

Many software tool developers, once they become big and successful, are want to over-confidence. They start forgetting the pivotal role that their power users had, and have, in maintaining the health of a software ecosystem, since the number of power users is often deceptively small.

If, in the expansionist search for a growing user base, you cater too much to contemporary gee-whizz features rather than the expressive power of **solid first principles,** you will find yourself overtaken in the long run.

Jesus won over Rome.
